body.tex

```diff
@@ -1,166 +1,466 @@
+
+
+The system at the end of the PoC is shown in Figure~\ref{fig:arch}.
+All components are hosted on the AWS platform.
+
+\begin{figure}
+  \centering
+  \includegraphics[width=\textwidth]{figures/arch}
+  \label{fig:arch}
+  \caption{Diagram of the AWS PoC system architecture.}
+\end{figure}
+
+AMIs containing LSST Software Stack, Pegasus, and configured HTCondor are pre-built in two flavors: master and worker. These contain the complete environment required to run DRP workflows and scale them out with HTCondor. The user launches an on-demand EC2 instance using the master AMI to be the submit host and the central manager of HTCondor pool.
+The user logs in to this master to control the workflow execution.
+
+The workflow of the LSST DRP in Generation 3 Middleware is represented as a Directed Acyclic Graph (DAG) known as Quantum Graph.
+In the Generation 3 Middleware, input and output datasets are defined in the Pipeline Tasks and each unit is known as a Quantum.
+The Quantum Graph is aware of the Quantum dependency and determines the execution workflow.
+For simplicity we map each Quantum to a standalone job, despite this may not be optimal for the performance.
+Pegasus is used to submit the workflow to HTCondor as well as monitor the workflow execution.
+Each LSST Pipeline Task Quantum is resource-independent. Pegasus adds other necessary jobs, such as data transfer, to the executable workflow. HTCondor DAGMan is the workflow execution engine behind Pegasus and controls the processes.
+
+Compute resources are procured from the master node via \texttt{condor\_annex} by targeting the worker AMI. Annex is a resource acquisition and external compute resource separation management unit. Each resource pool can have multiple annexes. Each annex manages its own lifecycle. It is possible to request on-demand or spot instance fleets.
+
+LSST Pipeline jobs are executed on the worker instances; they use Data Butler for data access.
+New Data Butler backends were implemented during the PoC, see Sect \ref{sec:butler}.
 
+The Butler datastore is located in an S3 Bucket and follows the same hierarchical structure that POSIX datastore does. Consumed and produced datasets are read and written directly from S3 as bytes, whenever possible, and only downloaded to temporary files for objects that are not serializable.
+Since the directory structure is preserved by the S3 datastore the entire data repository is trivially transferable between the cloud and a local filesystem.
 
+The Butler registry is a RDS PostgreSQL database that keeps track of all LSST science files.
+Files that are not managed by Data Butler are managed by the Condor File IO via Pegasus. These include the Data Butler configuration file, pipeline definition (Quantum) files, and the log files. The master instance serves as the staging site for these files. Various network protocols transfer the files between the master and the workers; instances do not share a filesystem.
```

Comment:
What are the network protocols for file transfer?

Reply:
A quick search in the Pegasus documents gave me: GridFTP, SCP, SRM, Amazon S3, Google Storage, XRootD, HTTP, and local cp/symlinking 

But I think they are not relevant here or we meant different Pegasus configurations. I'll rephrase.